packages/flutter/lib/src/services/text_formatter.dart

```diff
@@ -379,25 +379,22 @@ class LengthLimitingTextInputFormatter extends TextInputFormatter {
     TextEditingValue oldValue,
     TextEditingValue newValue,
   ) {
-    // Return the new value when the old value has not reached the max
-    // limit or the old value is composing too.
-    if (newValue.composing.isValid) {
-      if (maxLength != null && maxLength! > 0 &&
-          oldValue.text.characters.length == maxLength! &&
```

Comment:
This caused one of the tests to fail (line 5725 in `editable_text_test.dart`): https://github.com/flutter/flutter/blob/5b7869c1464c22673d98342d6c09213db3e53d0a/packages/flutter/test/widgets/editable_text_test.dart#L5725